flybywiresim/aircraft
# feat(EFB): Added Loadsheet/OFP Navigation

## Summary of Changes


- [x] **Feature:** Navigation through the different Loadsheet/OFP sections in the EFB, improving the reading speed and efficiency.
- [x] Support for OFP Layouts: [LIDO, AAL, ACA, AFR-2017, BAW, DAL, DLH, EIN, EZY, JBU, QFA, THY]
- [ ] Reset scroll after selecting a section.
- [ ] Reset the selected section after importing new simbrief data.
- [ ] Support for OFP Layouts: [UAL-2018, UAL-2012, KLM, BER, AFR-2012]
- [ ] Fix 'Maps', 'Map' or 'UAD Charts' section missing for all the OFP layouts.

## Screenshots (if necessary)



https://github.com/user-attachments/assets/d407a8ab-7cfe-47a7-bfdc-5b0d2cd1a61f

![Loadsheet Navigation with EZY OFP Layout](https://github.com/user-attachments/assets/e8cee49f-69b4-43d1-991e-7dd3a851d408)

## References


e.g LIDO Format:
![Screenshot 2024-11-11 181233](https://github.com/user-attachments/assets/f5e53dc3-cf90-4eef-bed3-6b111b6f0837)




## Additional context
Loadsheet navigation sections were extracted from a test dispatch made with each OFP layout.

From Simbrief we receive the Loadsheet in HTML format, which contains section-markers/comments like this e.g: 
- `` 

So, after getting a headache 😆 with early testing and doing some research, i finally came up with the following pattern:  
- `[\\s\\S]*?(?=|$)` 

to be  able to extract the content of each section.

## Testing instructions


1. Go to Simbrief and generate a flight (make sure to use the Official FBW A20N Profile).
2. Import the simbrief data through the EFB.
3. Select the **Dispatch** option in the EFB sidebar.
4. Select a desired section and ensure if the feature is working. 
5. Repeat all the steps with every OFP Layout.


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page
